core/src/main/java/org/apache/carbondata/core/keygenerator/columnar/impl/MultiDimKeyVarLengthEquiSplitGenerator.java

```diff
@@ -143,17 +142,18 @@ private void intialize() {
 
   private int[] convertToArray(List<Integer> list) {
```

Comment:
the whole purpose of this function is to unbox Integer array into integer array, not converting List[Integer] to Integer[]. 

 I looked closely at the caller of the function initialize(), which can be rewritten to avoid this back and forth set/list/array conversion.  and can be addressed in a separate PR.